modules/rtm/R/bayestools.R

```diff
@@ -67,98 +71,138 @@ prospect_bt_prior <- function(version, custom_prior = list()) {
-#'  * `init` -- BayesianTools settings for just the first round of sampling. 
-#'  This is most common for the initial number of iterations, which is the 
+#'  * `init` -- BayesianTools settings for just the first round of sampling.
+#'  This is most common for the initial number of iterations, which is the
 #'  minimum expected for convergence.
-#'  * `loop` -- BayesianTools settings for iterations inside the convergence 
-#'  checking `while` loop. This is most commonly for setting a smaller 
+#'  * `loop` -- BayesianTools settings for iterations inside the convergence
+#'  checking `while` loop. This is most commonly for setting a smaller
 #'  iteration count than in `init`.
 #'  * `other` -- Miscellaneous (non-BayesianTools) settings, including:
 #'      - `sampler` -- String describing which sampler to use. Default is `DEzs`
-#'      - `use_mpsrf` -- Use the multivariate PSRF to check convergence. 
-#'      Default is `FALSE` because it may be an excessively conservative 
+#'      - `use_mpsrf` -- Use the multivariate PSRF to check convergence.
+#'      Default is `FALSE` because it may be an excessively conservative
 #'      diagnostic.
 #'      - `min_samp` -- Minimum number of samples after burnin before stopping.
-#' 
+#'      Default is 1000.
+#'      - `max_iter` -- Maximum total number of iterations. Default is 1e6.
+#'      - `lag.max` -- Maximum lag to use for autocorrelation normalization.
+#'      Default is `10 * log10(n)` (same as `stats::acf` function).
+#'      - `save_progress` -- File name for saving samples between loop
+#'      iterations. If `NULL` (default), do not save progress samples.
+#'
 #' See the BayesianTools sampler documentation for what can go in the `BayesianTools` settings lists.
 #' @param observed Vector of observations
-#' @param model Function called by log-likelihood. Must be `function(params)` 
+#' @param model Function called by log-likelihood. Must be `function(params)`
 #' and return a vector equal to `length(observed)` or `nrow(observed)`.
 #' @param prior BayesianTools prior object.
 #' @param custom_settings Nested settings list. See Details.
 #' @export
 invert_bt <- function(observed, model, prior, custom_settings = list()) {
 
-    default_settings <- list(common = list(), 
+    default_settings <- list(common = list(),
                              init = list(iterations = 10000),
                              loop = list(iterations = 2000),
-                             other = list(sampler = 'DEzs', 
+                             other = list(sampler = 'DEzs',
                                           use_mpsrf = FALSE,
-                                          min_samp = 1000))
+                                          min_samp = 1000,
+                                          max_iter = 1e6,
+                                          lag.max = NULL,
+                                          save_progress = NULL))
```

Comment:
Not quite. `invert_bt(..., custom_settings = list(other = list(save_progress = "/path/to/file.rds")))`.